src/lib.rs

```diff
@@ -731,6 +733,38 @@ pub trait Future {
                         _>(f)
     }
 
+    /// Flatten the execution of this future when the successful result
+    /// of this future is a stream.
+    ///
+    /// This can be useful when stream initialization is deferred, and
+    /// it is convenient to work with that stream as if stream were
+    /// available at the call site.
+    ///
+    /// Note that this function consumes this future and returns a
+    /// wrapped version of it.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use futures::*;
+    /// use futures::stream::Stream;
+    ///
+    /// let future_of_a_stream = finished::<_, u32>(stream::iter(vec![
+    ///         Ok(17), Err(18), Ok(19)].into_iter()));
```

Comment:
> I think the into_iter may not be necessary here?

It is necessary, because signature of `iter` is:

```
pub fn iter<I, T, E>(i: I) -> IterStream<I>
    where I: Iterator<Item=Result<T, E>>
{ ... }
```

`I` is not `IntoIterator`.